src/managed/tools/bundle/bundle.cs

```diff
@@ -0,0 +1,630 @@
+    // For each embedded file, the following information is written:
+    //
+    // - - - - - - - - - - - - - - - - - - - - - - - -
+    // Name| File                   | File   | File
+    // Len | Name                   | Offset | Size
+    // - - - - - - - - - - - - - - - - - - - - - - - -
+
+    public class FileEntry
+    {
+        public FileType Type;
+        public string Name;
+        public long Offset;
+        public long Size;
+
+        public FileEntry(FileType fileType, string name, long offset, long size)
+        {
+            Type = fileType;
+            Name = name;
+            Offset = offset;
+            Size = size;
+        }
+
+        public void Write(BinaryWriter oneFile)
+        {
+            oneFile.Write(Name);
+            oneFile.Write(Offset);
+            oneFile.Write(Size);
+        }
+
+        public static FileEntry Read(BinaryReader oneFile, FileType fileType = FileType.Other)
+        {
+            string fileName = oneFile.ReadString();
+            long offset = oneFile.ReadInt64();
+            long size = oneFile.ReadInt64();
+            return new FileEntry(fileType, fileName, offset, size);
+        }
+
+        public override string ToString()
+        {
+            return String.Format($"{Name} [{Type}] @{Offset} Sz={Size}");
+        }
+    }
+
+    // BundleManifest:
+    // Here is the description of the Bundle Layout:
+    //_______________________________________________
+    //  AppHost 
+    //
+    //
+    // ----------------------------------------------
+    // The embedded files including the app, its
+    // configuration files, dependencies, and 
+    // possibly the runtime.
+    // 
+    // 
+    // 
+    // 
+    // 
+    // 
+    //
+    // -----------------------------------------------
+    // Meta-data Directory (FileEntrys)
+    //   "app.deps.json"           (if any)
+    //   "app.runtimeconfig.json"  (if any)
+    //   "app.dll"                 (The main app)
+    //   "IL_1.dll"                (Assemblies, if any)
+    //    ...
+    //   "IL_n.dll"
+    //   "File_1"                  (Other Files, if any)
+    //    ...
+    //   "File_n" 
+    // - - - - - - - - - - - - - - - - - - - - - - - -
+    //  Bundle Manifest (footer)
+    //     MajorVersion
+    //     MinorVersion
+    //     FileEntryStart (offset)
+    //     NumFiles
+    //     NumAssemblies
+    //     Flags
+    // -----------------------------------------------
+    //   Bundle Signature
+    // _________________________________________________
+
+    [Flags]
+    public enum BundleFlags
+    {
+        None = 0,
+        HasDepsJson = 1,
+        HasRuntimeConfigJson = 2
+    }
+
+    public class BundleManifest
+    {
+        public const string Signature = ".NetCoreBundle";
+        public const uint MajorVersion = 0;
+        public const uint MinorVersion = 1;
+
+        const uint SignatureLength = 15;
+        const uint FooterLength = 43;
```

Comment:
`FooterLength` includes `SignatureLength`, not just the size of the structure.

I think it is better to call out these constants for `SignatureLength` and `FooterLength`, because they will be communicated to the implementation of bundle parsing/extraction in native code in the `AppHost`.